### PR TITLE
Feat/filter functionality

### PR DESCRIPTION
**Pull request by AlexaVanSchaardenburg**
## What I did:
Built functionality to filter through parks by activity and handle any errors for bad routes, or 400 and 500 level errors on GET requests

## Did this break anything?
- [x] No
- [ ] Yes

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

## Checklist:
- [ ] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

## What's next?
Review and refactor my code and implement prop types before starting on cypress testing

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -1,56 +1,75 @@
...
       <Nav />
+      {error && <Navigate to='/error' />}
       <Routes>
-        {allParks && (
-          <Route path='/' element={<div>{allParksView(allParks)}</div>} />
-        )}
-        <Route path='/:parkCode' element={<Details allParks={allParks} />} />
+        <Route path='/' element={<div>{filteredParks ? homeView(filteredParks) : homeView(allParks)}</div>} />
+        <Route path='/:parkCode' element={<Details allParks={allParks} setError={setError} error={error}/>} />
+        <Route path='/error' />
```
Comment (mbenfowler) [inline src/App/App.js:71]: why do you have this route?

File: src/Loading/Loading.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import './Loading.css';
+import { NavLink } from 'react-router-dom'
+
+const Loading = () => {
+
+  return (
+      <div className='loading'>
+        <h1>Page is Loading...</h1>
+        {/* <NavLink to='/' className='go-home'>🏕️ go back to home 🏕️</NavLink> */}
```
Comment (mbenfowler) [inline src/Loading/Loading.js:9]: can probably remove this commented out code now? Unless there was something else you were wanting to try with it?

File: src/Form/Form.js [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
+        <option value="Guided Tours">guided tours</option>
+        <option value="Hiking">hiking</option>
+        <option value="Front Country Hiking">front-country hiking</option>
+        <option value="Junior Ranger Program">junior ranger program</option>
+        <option value="Wildlife Watching">wildlife watching</option>
+        <option value="Birdwatching">birdwatching</option>
+        <option value="Park Film">park film</option>
+        <option value="Museum Exhibits">museum exhibits</option>
+        <option value="Shopping">shopping</option>
+        <option value="Bookstore and Park Store">bookstore and park store</option>
```
Comment (mbenfowler) [inline src/Form/Form.js:36]: You could probably save all these activities to a data file somewhere and `map` through them here in your `Form` component - would clean it up a ton and make it easier to update if you need to add/remove activities from the list.

File: src/Error/Error.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import './Error.css';
+import { NavLink } from 'react-router-dom';
+
+const Error = ({ error, view }) => {
+
+
+
```
Comment (mbenfowler) [inline src/Error/Error.js:7]: remove extra lines

File: src/App/App.js [hunk 5]
```diff
@@ -1,56 +1,75 @@
...
   return (
     <>
       <Nav />
+      {error && <Navigate to='/error' />}
       <Routes>
-        {allParks && (
-          <Route path='/' element={<div>{allParksView(allParks)}</div>} />
-        )}
-        <Route path='/:parkCode' element={<Details allParks={allParks} />} />
+        <Route path='/' element={<div>{filteredParks ? homeView(filteredParks) : homeView(allParks)}</div>} />
```
Comment (mbenfowler) [inline src/App/App.js:69]: nice!!

File: src/Details/Details.js [hunk 6]
```diff
@@ -69,7 +70,7 @@ const Details = () => {
             )
           })}
       </div>
-      : <p>loading ...</p>}
+      : <Error error={error}/>}
```
Comment (mbenfowler) [inline src/Details/Details.js:73]: I'm seeing this render briefly on initial load of main page and details view - you may also want a state of `isLoading` that will fallback to a loading view until the data is ready to be displayed.

**Review by mbenfowler — CHANGES_REQUESTED** [review]

File: src/App/App.js (hunk 1 above)
Comment (AlexaVanSchaardenburg) [reply src/App/App.js:71]: lol, I am not sure. I took it out

File: src/Form/Form.js (hunk 3 above)
Comment (AlexaVanSchaardenburg) [reply src/Form/Form.js:36]: Great idea!

File: src/App/App.js (hunk 5 above)
Comment (AlexaVanSchaardenburg) [reply src/App/App.js:69]: Thank you!

**Pull request merged**